Entities/MOSRotating.h

```diff
@@ -136,16 +136,36 @@ ClassInfoGetters
 
     void Destroy(bool notInherited = false) override;
 
+    /// <summary>
+    /// Gets the mass value of this MOSRotating, including the mass of all its Attachables and their Attachables and so on.
+    /// </summary>
+    /// <returns>The mass of this MOSRotating and all Attachables in Kilograms (kg).</returns>
+    float GetMass() const override { return MOSprite::GetMass() + m_AttachableAndWoundMass; }
```

Comment:
Good point on `MovableObject::GetMass();`.

As for the other stuff, I disagree. Seems to me anything that wants to know an object's mass will want to know its actual mass, not just the mass of that one part. This keeps everything very consistent - GetMass will always return the actual mass of the object, no need to remember to use GetTotalMass for some stuff and GetMass for others.
If there's a need for getting just the object's mass, I'd rather have a separate method for that, or an optional parameter for getMass that specifies to only get the object's mass instead of its actual mass.

Reply:
Fair enough, no need for an extra parameter either because this is easily solvable for the lua side of things where this is the main concern.
You forgot to add a binding for it, so add one to MOSR and make it a function call and call it `GetTotalMass` since you can name the bindings anything you want.
The property binding is under MO and calls `MO::GetMass` and that returns just this object's mass.
Just make sure the numbers they return are correct in case it somehow screws up because of the override and gets you `MOSR::GetMass` for the property one.

Reply:
Everything seems fine just using the mo.Mass binding. Also, I'm not sure I get the point of having a GetTotalMass binding really, for the reasons I mentioned above. Let's discuss it in discord.